database/redis/database.go

```diff
@@ -108,6 +108,12 @@ func NewDatabase(logger moira.Logger, config Config, source DBSource) *DbConnect
 	return connector
 }
 
+// NewTestDatabase use it only for tests
+func NewTestDatabase(logger moira.Logger, config Config, source DBSource) *DbConnector {
```

Comment:
I think you should hardcode `source` here, and not pass it as a parameter. It should always be the same for tests.

Reply:
fixed